### PR TITLE
Refactor docs to improve clarity around connector development.

### PR DESCRIPTION
**Pull request by avaidyanatha**
## Main Changes
- Moves `Connector Development` in to its own top level category for clarity
- Unifies one `Tutorials` section for all connector development
- Moves `A Beginner's Guide to the AirbyteCatalog` to `Understanding Airbyte`

## Secondary Changes
- Moves Monorepo development to `Contributing to Airbyte`
- Removes the `Java Connectors Doc`
- Removes the `Building a Connector the Hard Way` and `Adding Incremental to a Source` docs
- Other QOL changes

### CODE REVIEW

File: docs/connector-development/README.md [hunk 1]
```diff
@@ -54,15 +54,15 @@ and choose the relevant template. This will generate a new connector in the `air
 
 Search the generated directory for "TODO"s and follow them to implement your connector. For more detailed walkthroughs and instructions, follow the relevant tutorial:
 
-* [Building a Python source ](../tutorials/building-a-python-source.md) 
-* [Building a Python destination](../tutorials/building-a-python-destination.md)
-* [Building a Java destination ](../tutorials/building-a-java-destination.md)
+* [Building a Python source ](tutorials/building-a-python-source.md)
```
Comment (michel-tricot) [inline docs/connector-development/README.md:57]: @sherifnada shouldn't we also have smth about the CDK here?

File: docs/connector-development/airbyte101.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Airbyte 101 for Connector Development
```
Comment (michel-tricot) [inline docs/connector-development/airbyte101.md:1]: What is the purpose of that page?

File: docs/connector-development/cdk-python/README.md [hunk 3]
```diff
@@ -25,7 +25,25 @@ Additionally, you can follow [this tutorial](https://github.com/airbytehq/airbyt
...
+
+Having trouble figuring out how to write a `stream_slices` function or aren't sure what a `cursor_field` is? Head to our [incremental stream doc](../incremental-stream.md).
+
+#### Practical Tips
+
+Airbyte recommends using the CDK template generator to develop with the CDK. The template generates created all the required scaffolding, with convenient TODOs, allowing developers to truly focus on implementing the API.
+
+For tips on useful Python knowledge, see the [Python Concepts](../python-concepts.md) page.
+
+You can find a complete tutorial for implementing an HTTP source connector in [this tutorial](https://github.com/airbytehq/airbyte/tree/4a397d25247db77a7b78783d26dae35bc3900f59/airbyte-cdk/python/docs/tutorials/http_api_source.md)
```
Comment (michel-tricot) [inline docs/connector-development/cdk-python/README.md:46]: wrong file

**Review by michel-tricot — COMMENTED** [review]
I'll let @sherifnada approve! Thanks for the change

File: docs/connector-development/README.md (hunk 1 above)
Comment (avaidyanatha) [reply docs/connector-development/README.md:57]: There's a note about the CDK at the top of the document, but I can definitely reiterate here. I'll let @sherifnada respond.

File: docs/connector-development/airbyte101.md (hunk 2 above)
Comment (avaidyanatha) [reply docs/connector-development/airbyte101.md:1]: @sherifnada and I decided that we need a good home for concepts that are useful to understand when developing connectors. We needed a way to link to the `Beginner's guide to Catalog` doc and this is the way that we came up with.

File: docs/connector-development/cdk-python/README.md (hunk 3 above)
Comment (avaidyanatha) [reply docs/connector-development/cdk-python/README.md]: Fixed, along with some others.

File: docs/connector-development/README.md (hunk 1 above)
Comment (sherifnada) [reply docs/connector-development/README.md:56]: can't hurt to add a line like `most of these tutorials demonstrate how to use the [CDK](LINK) to build a connector`

File: docs/contributing-to-airbyte/building-new-connector/legacy-standard-source-tests.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 ## Airbyte's Standard Tests (v1)
```
Comment (sherifnada) [inline docs/contributing-to-airbyte/building-new-connector/legacy-standard-source-tests.md:3]: should we also move this page to the new connector dev section?

**Review by sherifnada — APPROVED** [review]

File: docs/contributing-to-airbyte/building-new-connector/legacy-standard-source-tests.md (hunk 4 above)
Comment (avaidyanatha) [reply docs/contributing-to-airbyte/building-new-connector/legacy-standard-source-tests.md:3]: Done.

File: docs/connector-development/README.md (hunk 1 above)
Comment (avaidyanatha) [reply docs/connector-development/README.md]: Added two links to the CDK tutorials!

**Comment by etsybaev** [comment]
Hi @avaidyanatha.
I just pulled the latest master and found that I can't build ht project anymore.
For example, after running such "SUB_BUILD=CONNECTORS_BASE ./gradlew clean build" command if fails like this:
Selection_286


Seems to be related that migrated files from this PR (please correct me if I'm wrong).
Could you please help there? Do we need to build a project in another way now?
Thanks in advance!

**Comment by avaidyanatha** [comment]
> Hi @avaidyanatha.
> I just pulled the latest master and found that I can't build ht project anymore.
> For example, after running such "SUB_BUILD=CONNECTORS_BASE ./gradlew clean build" command if fails like this:
> Selection_286
> 
> Seems to be related to migrated files from this PR (please correct me if I'm wrong).
> Could you please help there? Do we need to build a project in another way now?
> Thanks in advance!

I see that `build.gradle` file was recently updated with the root directory, is your problem solved now?

**Pull request merged**